projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -1326,7 +1329,14 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
      *@hidden
      */
     private getDateOnlyInMs(date: Date) {
-        return new Date(date.getFullYear(), date.getMonth(), date.getDate()).getTime();
+        return this.getDateOnly(date).getTime();
+    }
+
+     /**
+     *@hidden
+     */
+    private getDateOnly(date: Date) {
+        return new Date(date.getFullYear(), date.getMonth(), date.getDate());
```

Comment:
This is an issue. If I have a date object that I bind both date and time picker to, then the date picker will reset the values that have been set through the time picker.

Reply:
Actually it won't, but this does not mean that we've implemented this functionality correctly.

Here goes why. `getDateOnly` is called on igxDatePicker opening, this is the `date` that will be presented in the default view. `this.calendar.viewDate` by default will be the current date (without the `time` part), and when a new date is selected `getDateOnly` will change the `viewDate` to be the new selected date (without the `time` part).

In `handleSelection` the time portion is being added, and `viewDate` updated accordingly. So it looks like we have inconsistency, because sometimes the date is with the time portion and sometimes it is not.

Either way we need to review our approach.

Reply:
The changes in the calendar does not affect the datePicker.
We had an issue related to time part when select date from datePicker #2561. It is fixed and has test.